contracts/liquidity_hub/pool-network/terraswap_pair/src/commands.rs

```diff
@@ -403,15 +403,18 @@ pub fn swap(
         amount: swap_computation.return_amount,
     };
 
+    let fees = swap_computation
+        .swap_fee_amount
+        .checked_add(swap_computation.protocol_fee_amount)?
+        .checked_add(swap_computation.burn_fee_amount)?;
     // check max spread limit if exist
+
     helpers::assert_max_spread(
         belief_price,
         max_spread,
-        offer_asset.clone(),
-        return_asset.clone(),
+        offer_asset.amount,
+        return_asset.amount.checked_add(fees)?,
```

Comment:
fees should be accounted for when calculating spread in the return amount, otherwise the max_spread input by the user is gonna be off